src/client/datascience/commands/activeEditorContext.ts

```diff
@@ -179,25 +185,30 @@ export class ActiveEditorContextService implements IExtensionSingleActivationSer
         }
         this.updateContextOfActiveNotebookKernel(activeEditor);
     }
-    private async updateDebugContext(activeEditor?: INotebookEditor, interpreter?: PythonEnvironment) {
-        this.canDebug.set(false).ignoreErrors();
-
-        if (activeEditor && activeEditor === this.notebookEditorProvider.activeEditor) {
-            const nb = await this.notebookProvider.getOrCreateNotebook({
-                identity: activeEditor.file,
-                resource: activeEditor.file,
-                getOnly: true
-            });
-            const interpreter = nb?.getMatchingInterpreter();
-            if (interpreter) {
-                this.canDebug
-                    .set(await this.dependencyService.areDebuggingDependenciesInstalled(interpreter))
-                    .ignoreErrors();
+    private onNotebookControllerSelected({
+        controller
+    }: {
+        notebook: NotebookDocument;
+        controller: VSCodeNotebookController;
+    }) {
+        this.updateDebugContext(controller.connection.interpreter);
```

Comment:
I think that you also need a check on if this is the same as the activeNotebookEditor document. It would usually be the case but there isn't a guarantee that this message is coming in from the active document. You might accidentally update with the interpreter from a different controller on a non-active notebook editor.